typescript-test-samples/step-functions-local/src/tests/sfnLocal/step-functions-local.test.ts

```diff
@@ -0,0 +1,183 @@
+ * the step function state machine. 
+ * 
+ */
+
+describe('lead-generation', () => {
+    describe('step-function-local', () => {
+
+        let container: StartedTestContainer;
+        let sfnClient: SFNClient;
+        let createStateMachineResponse: CreateStateMachineCommandOutput;
+        let stateMachineArn: string;
+
+        beforeAll(async () => {
+
+            // Launch StepFunctionLocal with testcontainers
+            container = await new GenericContainer('amazon/aws-stepfunctions-local')
+                .withExposedPorts(8083)
+                .withCopyFilesToContainer([{ 
+                    source: StepFunctionsConstants.mockFileHostPath, 
+                    target: StepFunctionsConstants.mockFileContainerPath
+                  }])
+                .withEnvironment({ SFN_MOCK_CONFIG: StepFunctionsConstants.mockFileContainerPath })
+                .withDefaultLogDriver()
+                .withWaitStrategy(Wait.forLogMessage("Starting server on port 8083"))
+                .start();
+
+            // Set up Step Function client with test container URL
+            sfnClient = new SFNClient({ 
+                endpoint: `http://${container.getHost()}:${container.getMappedPort(8083)}`
+            });                
+
+            // Important to avoid non-deterministic behavior
+            await sleep(2);         
+
+            // Create state machine
+            createStateMachineResponse = await sfnClient.send(new CreateStateMachineCommand({
+                name: StepFunctionsConstants.STATE_MACHINE_NAME,
+                definition: StepFunctionsConstants.STATE_MACHINE_ASL,
+                roleArn: StepFunctionsConstants.DUMMY_ROLE
+            }));
+
+            stateMachineArn = createStateMachineResponse.stateMachineArn as string;
+
+        });
+        
+        afterAll(async () => {
+            // stop the SFN client
+            await container.stop();
+        });
+
+        /**
+         * Unit tests.
+         *
+         * Now we come to the unit tests themselves. Each test checks a single
+         * scenario, with a new state machine execution.
+         *
+         */
+
+        it('Check Container is running', () => {
+            expect(container.getId()).toBeDefined();
+        });
+
+        it('Test Happy Path Scenario', async () => {
+            const executionName: string = "happyPathExecution";
+            const executionResponse: StartExecutionCommandOutput = await sfnClient.send(new StartExecutionCommand({
+                name: executionName,
+                stateMachineArn: `${stateMachineArn}#HappyPathTest`,
+                input: StepFunctionsConstants.EVENT_JSON_STRING
+            }));
+            
+            expect(executionResponse.executionArn).not.toBeNull();
+            
+            // IMP: Wait until above execution completes in docker
+            await sleep(2);
+
+            const historyResponse = await sfnClient.send(new GetExecutionHistoryCommand({
```

Comment:
Is it worth looping here to check execution results?

Reply:
Added expect(historyResponse).toBeDefined(); 
Proper error handling is out of scope for these code samples on unit testing, right? Otherwise I'll look further into this.

Reply:
Yes, production-level error handling is out of scope.

For this case, I'm focused on making the sample reliable because unexpected failures tend to throw off folks who are using the samples to get ramped up. 

If 2 seconds is consistently enough time for the process to complete, the current code should be sufficient. If not, is there a way to make it more reliable? Or, if not that could we build a more explicit error message to report what happened and what the tester can do about it (maybe just re-run the test)?

Reply:
I'll look into this.

Reply:
I've bundled this into a function that loops until the execution completes. I've tested with 100ms delay and it works (i takes around 1s to complete)